manifest-build-tools/docker_build.sh

```diff
@@ -1,33 +1,57 @@
-#!/bin/bash
+#!/bin/bash 
+set -e
 
 #
 # This script is for build/push rackhd docker images.
 # Environmental requirement:
-#   1.docker service running and docker have already logged with Rackhd Dockerhub ID, 
-#     cmd 'docker login', if not logged then can't push images to dockerhub
-#   2.A sources.list to replace default sources.list of nodesource/wheezy:4.4.6 image(the root image which on-core pulled)
+#   1.A sources.list to replace default sources.list of nodesource/wheezy:4.4.6 image(the root image which on-core pulled)
```

Comment:
clone all repos and checkout the specific commits , are also the pre-condition of this script , right ?